google/cloud/aiplatform/metadata/resource.py

```diff
@@ -0,0 +1,300 @@
+
+        Args:
+            resource_id (str):
+                Required. The <resource_id> portion of the resource name with
+                the format:
+                projects/123/locations/us-central1/metadataStores/<metadata_store_id>/<resource_noun>/<resource_id>.
+            schema_title (str):
+                Required. schema_title identifies the schema title used by the resource.
+            display_name (str):
+                Optional. The user-defined name of the resource.
+            schema_version (str):
+                Optional. schema_version specifies the version used by the resource.
+                If not set, defaults to use the latest version.
+            description (str):
+                Optional. Describes the purpose of the resource to be created.
+            metadata (Dict):
+                Optional. Contains the metadata information that will be stored in the resource.
+            metadata_store_id (str):
+                The <metadata_store_id> portion of the resource name with
+                the format:
+                projects/123/locations/us-central1/metadataStores/<metadata_store_id>/<resource_noun>/<resource_id>
+                If not provided, the MetadataStore's ID will be set to "default".
+            project (str):
+                Project used to create this resource. Overrides project set in
+                aiplatform.init.
+            location (str):
+                Location used to create this resource. Overrides location set in
+                aiplatform.init.
+            credentials (auth_credentials.Credentials):
+                Custom credentials used to create this resource. Overrides
+                credentials set in aiplatform.init.
+
+        Returns:
+            resource (_Resource):
+                Instantiated representation of the managed Metadata resource.
+
+        """
+        api_client = cls._instantiate_client(location=location, credentials=credentials)
+
+        parent = (
+            initializer.global_config.common_location_path(
+                project=project, location=location
+            )
+            + f"/metadataStores/{metadata_store_id}"
+        )
+
+        try:
+            cls._create_resource(
+                client=api_client,
+                parent=parent,
+                resource_id=resource_id,
+                schema_title=schema_title,
+                display_name=display_name,
+                schema_version=schema_version,
+                description=description,
+                metadata=metadata,
+            )
+        except exceptions.AlreadyExists:
+            logging.info(f"Resource '{resource_id}' already exist")
+
+        return cls(
+            resource_name=f"{parent}/{cls._resource_noun}/{resource_id}",
+            metadata_store_id=metadata_store_id,
+            project=project,
+            location=location,
+            credentials=credentials,
+        )
+
+    @classmethod
+    def get(
```

Comment:
This seems redundant with the constructor. Why do we need a get API with the constructor which also gets?

Reply:
After reviewing the code up the stack, I think this API should be a `get_or_create` instead of pushing that logic to the caller.

Reply:
I tried using the constructor but I wasn't able to properly catch exceptions... Also from the caller, it felt strange to do something like:

`Execution(execution_name="my-execution")`, which was actually doing an RPC call. 

Let me see if I can change it to get_or_create, and make the create and get private...

Reply:
Only exposing get_or_create and update now. Thanks for the suggestion, caller code is much cleaner now